src/persist-client/src/batch.rs

```diff
@@ -898,16 +898,58 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
         let index = u64::cast_from(self.finished_parts.len() + self.writing_parts.len());
         let ts_rewrite = None;
 
-        let handle = if updates.goodbytes() < self.cfg.inline_writes_single_max_bytes {
+        let should_columnar_encode = self.cfg.batch_columnar_format.is_structured()
```

Comment:
Maybe make `should_columnar_encode` a method on `BatchBuilderConfig` to deduplicate it with the other one? It's only a tiny bit of logic, but feels like the type of thing that could easily rot in one path.

Reply:
Good idea, updated!